rct_image_tools/src/rct_image_tools/modified_circle_grid_finder.cpp

```diff
@@ -106,38 +101,94 @@ static std::vector<cv::Point2d> extractKeyPoints(const cv::Mat& image, const std
   std::size_t temp_rows = flipped ? cols : rows;
   std::size_t temp_cols = flipped ? rows : cols;
 
+  std::size_t start_first_row = 0;
+  std::size_t end_first_row = cols - 1;
+  std::size_t start_last_row = rows * cols - cols;
+  std::size_t end_last_row = rows * cols - 1;
+
   // Determine which circle is the largest
   double start_first_row_size = -1.0;
   double start_last_row_size = -1.0;
   double end_first_row_size = -1.0;
   double end_last_row_size = -1.0;
 
+  // Determine size of corners relative to their adjacent circles
+  double start_first_row_avg_rel_size = 0;
+  double start_last_row_avg_rel_size = 0;
+  double end_first_row_avg_rel_size = 0;
+  double end_last_row_avg_rel_size = 0;
+
   cv::Point2d start_last_row_pt = centers[start_last_row];
   cv::Point2d end_last_row_pt = centers[end_last_row];
   cv::Point2d start_first_row_pt = centers[start_first_row];
   cv::Point2d end_first_row_pt = centers[end_first_row];
 
+  // Get the size of every circle
+  std::vector<double> center_sizes(centers.size(), 1.0);
```

Comment:
The name `center_sizes` is a bit misleading. `centers` is an argument that defines the locations of the features in the calibration target. These features have an actual size on the pattern. However, this particular variable actually represents the size of the observed keypoint features that correspond to the features on the calibration target. I would suggest renaming the argument `centers` to `target_features` and renaming to something like `obs_feature_sizes` to be more clear. It's a little outside the scope of this PR, but if we're making changes, we should make them clear

Reply:
Renamed `centers` to `feature_coordinates` and `center_sizes` to `feature_sizes`, this mimics the keypoints definitions more closely because in  the `KeyPoint` struct they refer to the `pt` field as the "coordinates of the keypoints".